src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -0,0 +1,428 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { LitElement, html, customElement, property } from 'lit-element';
+import * as MicrosoftGraph from '@microsoft/microsoft-graph-types';
+
+import { Providers } from '../../Providers';
+import { ProviderState } from '../../providers/IProvider';
+import { styles } from './mgt-people-picker-css';
+
+import '../mgt-person/mgt-person';
+import '../../styles/fabric-icon-font';
+import { MgtTemplatedComponent } from '../templatedComponent';
+import { MgtPersonDetails, MgtPerson } from '../mgt-person/mgt-person';
+
+@customElement('mgt-people-picker')
+export class MgtPicker extends MgtTemplatedComponent {
+  private _firstUpdated = false;
+  private groupPeople;
+  private isLoading = false;
+
+  @property({
+    attribute: 'people',
+    type: Object
+  })
+  people: Array<MgtPersonDetails> = null;
+
+  @property({
+    attribute: 'show-max',
+    type: Number
+  })
+  showMax: number = 6;
+
+  private arrowSelectionCount: number = 0;
+
+  @property({
+    attribute: 'group',
+    type: String
+  })
+  group: string;
+
+  @property() private _personName: string = '';
+  @property() private _selectedPeople: Array<any> = [];
+  @property() private _duplicatePersonId: string = '';
+  @property() private _userInput: string = '';
+
+  static get styles() {
+    return styles;
+  }
+
+  constructor() {
+    super();
+    this.trackMouseFocus = this.trackMouseFocus.bind(this);
+  }
+
+  firstUpdated() {
+    if (this.group) {
+      Providers.onProviderUpdated(() => this.findGroup());
+      this.findGroup();
+    }
+  }
+
+  private async findGroup() {
+    let provider = Providers.globalProvider;
+    if (provider && provider.state === ProviderState.SignedIn) {
+      let client = Providers.globalProvider.graph;
+      this.groupPeople = await client.getPeopleFromGroup(this.group);
+    }
+  }
+
+  private onUserTypeSearch(event: any) {
+    if (event.keyCode == 40 || event.keyCode == 38) {
+      return;
+    }
+    if (event.code == 'Escape') {
+      event.target.value = '';
+      this._userInput = '';
+      this.people = [];
+      return;
+    }
+    if (event.code == 'Backspace' && this._userInput.length == 0 && this._selectedPeople.length > 0) {
+      event.target.value = '';
+      this._userInput = '';
+      //remove last person in selected list
+      this._selectedPeople = this._selectedPeople.splice(0, this._selectedPeople.length - 1);
+      //fire selected people changed event
+      this.fireCustomEvent('selectionChanged', this._selectedPeople);
+      return;
+    }
+    this._userInput = event.target.value;
+    if (event.target.value) {
+      window.setTimeout(() => {
```

Comment:
need to clear timeout - create a debounce function in utils.js and use here

Reply:
Here's the debounce function: 
https://github.com/microsoftgraph/microsoft-graph-toolkit/commit/dc8dca87b40f7fce4c6f67a710ad9d3bccae3004

quick fix :
https://github.com/microsoftgraph/microsoft-graph-toolkit/commit/cf9c71540ce81123aa9915ad67a916d636492f80

Reply:
Much better version: 
https://github.com/microsoftgraph/microsoft-graph-toolkit/commit/0d22ec6a8da3c2cb53c3f0b927ff429f706d229c